src/NServiceBus.Gateway/Receiving/SingleCallChannelReceiver.cs

```diff
@@ -31,15 +31,9 @@ public void Start(Channel channel, int maxConcurrency, Func<MessageReceivedOnCha
             channelReceiver.Start(channel.Address, maxConcurrency, DataReceivedOnChannel);
         }
 
-        public void Dispose()
-        {
-            //Injected at compile time
-            DisposeManaged();
-        }
-
-        void DisposeManaged()
-        {
-            channelReceiver?.Dispose();
+        public Task Stop()
+        { 
+            return channelReceiver?.Stop();
```

Comment:
Elvis doesn't work with `await`. I have no idea if it works here. Can you verify?